### PR TITLE
Improve gRPC runner error handling

### PR DESCRIPTION
**Pull request by mxsdev**
Significantly improves the error handling situation for gRPC runner. Included is:

 - Handle case where server is disconnected prematurely and close terminal (previously terminal task would hang)
 - If grpc runner fails, show error message to user
 - If grpc environment creation fails, show error message to user

### CODE REVIEW

File: src/extension/kernel.ts [hunk 1]
```diff
@@ -323,17 +324,22 @@ export class Kernel implements Disposable {
...
+            // later we might want a more sophisticated approach/to bring this serverside
+            Object.entries(process.env).map(([k, v]) => `${k}=${v || ''}`)
+          )
+
           if(this.runner !== runner) { return }
           this.environment = env
-        })
+        } catch(e: any) {
+          window.showErrorMessage(`[Runme] Failed to create environment for gRPC Runner: ${e.message}`)
+          console.error('Failed to create gRPC Runner environment', e)
```
Comment (christian-bromann) [inline src/extension/kernel.ts:341]: I've seen `[Runme] ` prefixes mostly in our internal logs, no strong preference though, feel free to disregard.

```suggestion
          window.showErrorMessage(`Failed to create environment for gRPC Runner: ${e.message}`)
          console.error('[Runme] Failed to create gRPC Runner environment', e)
```

File: src/extension/runner.ts [hunk 2]
```diff
@@ -440,6 +451,13 @@ export class GrpcRunnerProgramSession implements IRunnerProgramSession {
   }
 
   open(): void {
+    // Workaround to force terminal to close if opened after early exit
+    // TODO(mxs): find a better solution here
```
Comment (christian-bromann) [inline src/extension/runner.ts:455]: Should we port this into an issue before we land the PR?

File: src/extension/kernel.ts [hunk 3]
```diff
@@ -277,7 +277,8 @@ export class Kernel implements Disposable {
         environmentManager
       )
         .catch((e) => {
-          console.error(e)
+          window.showErrorMessage(`Internal failure executing runner: ${e.message}`)
+          console.error('Internal failure executing runner', e.message)
```
Comment (christian-bromann) [inline src/extension/kernel.ts:281]:
```suggestion
          console.error('[Runme] Internal failure executing runner', e.message)
```

File: src/extension/kernel.ts (hunk 1 above)
Comment (christian-bromann) [reply src/extension/kernel.ts:341]: Screenshot 2023-03-14 at 14 06 55

**Review by christian-bromann — COMMENTED** [review]
Some minor comments, otherwise LGTM 👍

File: src/extension/runner.ts (hunk 2 above)
Comment (mxsdev) [reply src/extension/runner.ts:455]: I'll add an issue for it.

I don't think it's very high priority, basically just the `_onDidClose` gets fired twice for an interactive task. I think most things that listen to that are idempotent so it should be fine.

Comment (mxsdev) [reply src/extension/runner.ts:455]: #286

**Pull request merged**